### PR TITLE
Allow orders to be filtered by number

### PR DESCRIPTION
**Pull request by SzymJ**
I want to merge this change because it adds new filter `numbers` for orders query.

- is not allowed to use `ids` and `numbers` in the same filter
- `number` field has index in the database

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-7195 Allow clients with MANAGE_ORDERS to filter orders by number](https://app.clickup.com/t/2549495/SALEOR-7195)

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ All notable, unreleased changes to this project will be documented in this file.
 - Fix incorrect default value used in `PaymentInput.storePaymentMethod` - #9943 by @korycins
 - Introduce plain text attribute - #9907 by @IKarbowiak
 - Stop auto-assigning default addresses to checkout - #9933 by @SzymJ
+- `numbers` filter added to `OrderFilter` - #9967 by @SzymJ
```
Comment (maarcingebala) [inline CHANGELOG.md:12]:
```suggestion
- Added `OrderFilter.numbers` filter - #9967 by @SzymJ
```

File: saleor/graphql/order/filters.py [hunk 2]
```diff
@@ -190,7 +198,15 @@ class OrderFilter(DraftOrderFilter):
     ids = GlobalIDMultipleChoiceFilter(method=filter_order_by_id)
     gift_card_used = django_filters.BooleanFilter(method=filter_gift_card_used)
     gift_card_bought = django_filters.BooleanFilter(method=filter_gift_card_bought)
+    numbers = ListObjectTypeFilter(input_class=graphene.Int, method=filter_by_name)
```
Comment (maarcingebala) [inline saleor/graphql/order/filters.py:201]:
```suggestion
    numbers = ListObjectTypeFilter(input_class=graphene.Int, method=filter_by_order_number)
```

File: saleor/graphql/order/filters.py [hunk 3]
```diff
@@ -154,6 +156,12 @@ def filter_order_by_id(qs, _, value):
     return qs.filter(Q(id__in=pks) | (Q(use_old_id=True) & Q(number__in=old_pks)))
 
 
+def filter_by_name(qs, _, values):
```
Comment (maarcingebala) [inline saleor/graphql/order/filters.py:159]:
```suggestion
def filter_by_order_number(qs, _, values):
```

File: saleor/graphql/order/tests/test_order.py [hunk 4]
```diff
@@ -9159,6 +9159,58 @@ def test_orders_query_with_filter_charge_status(
...
+
+    # when
+    response = staff_api_client.post_graphql(
+        orders_query_with_filter, variables, permissions=(permission_manage_orders,)
+    )
+
+    # then
+    content = get_graphql_content(response)
+    orders = content["data"]["orders"]["edges"]
+    assert len(orders) == 2
```
Comment (maarcingebala) [inline saleor/graphql/order/tests/test_order.py:9184]: I would also check that the order numbers we got in the response are in fact the same order numbers we passed to the filter.

**Review by maarcingebala — APPROVED** [review]

File: saleor/graphql/order/filters.py [hunk 5]
```diff
@@ -190,7 +198,17 @@ class OrderFilter(DraftOrderFilter):
     ids = GlobalIDMultipleChoiceFilter(method=filter_order_by_id)
     gift_card_used = django_filters.BooleanFilter(method=filter_gift_card_used)
     gift_card_bought = django_filters.BooleanFilter(method=filter_gift_card_bought)
+    numbers = ListObjectTypeFilter(
+        input_class=graphene.Int, method=filter_by_order_number
```
Comment (patrys) [inline saleor/graphql/order/filters.py:202]: https://github.com/saleor/saleor/blob/b79781377120088769cebc11048b2db800d397c8/saleor/graphql/schema.graphql#L8066-L8067

**Review by patrys — APPROVED** [review]

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**